### PR TITLE
Move RASP sources to repository, Replace FileList with DownloadFIlePicker

### PR DESCRIPTION
**Pull request by lordfolken**

### CODE REVIEW

File: src/Repository/FileType.hpp [hunk 1]
```diff
@@ -33,4 +33,5 @@ enum class FileType : uint8_t {
   MAP,
   FLARMNET,
   IGC,
+  RASP
```
Comment (MaxKellermann) [inline src/Repository/FileType.hpp:36]: Missing trailing comma

File: src/Dialogs/Weather/RASPDialog.cpp [hunk 2]
```diff
@@ -22,29 +22,20 @@ Copyright_License {
...
 #include "ActionInterface.hpp"
 #include "Language/Language.hpp"
 #include "LocalPath.hpp"
-#include "Operation/PluggableOperationEnvironment.hpp"
-#include "co/InvokeTask.hxx"
-#include "net/http/Init.hpp"
-#include "net/http/CoDownloadToFile.hpp"
+
+#include "io/FileLineReader.hpp"
+#include <tchar.h>
```
Comment (MaxKellermann) [inline src/Dialogs/Weather/RASPDialog.cpp:38]: Why?

File: NEWS.txt [hunk 3]
```diff
@@ -39,10 +41,10 @@ Version 7.24 - 2022/07/22
   - rework menu labels
   - increase popup message timeouts
   - fix missing wind arrow InfoBox updates
-* file handling
```
Comment (MaxKellermann) [inline NEWS.txt:42]: Whitespace change?

File: NEWS.txt [hunk 4]
```diff
@@ -1,4 +1,6 @@
 Version 7.27 - not yet released
+* user interface
+  - rasp sources from repository
```
Comment (MaxKellermann) [inline NEWS.txt:3]: NEWS changes should be part of the commit implementing the feature; that way, a "git blame" on NEWS.txt easily reveals the commit intruducing the change.

File: src/Dialogs/Weather/RASPDialog.cpp [hunk 5]
```diff
@@ -22,29 +22,20 @@ Copyright_License {
...
-#include "Form/Edit.hpp"
 #include "Form/DataField/Enum.hpp"
 #include "Form/DataField/Listener.hpp"
-#include "Protection.hpp"
 #include "DataGlobals.hpp"
 #include "UIGlobals.hpp"
 #include "UIState.hpp"
 #include "ActionInterface.hpp"
 #include "Language/Language.hpp"
 #include "LocalPath.hpp"
```
Comment (MaxKellermann) [inline src/Dialogs/Weather/RASPDialog.cpp:35]: Can be removed as well

File: src/Dialogs/Weather/RASPDialog.cpp [hunk 6]
```diff
@@ -22,29 +22,20 @@ Copyright_License {
...
-#include "Dialogs/CoDialog.hpp"
-#include "Dialogs/Error.hpp"
-#include "Renderer/TextRowRenderer.hpp"
+#include "Dialogs/DownloadFilePicker.hpp"
 #include "Widget/RowFormWidget.hpp"
-#include "Look/DialogLook.hpp"
 #include "Weather/Rasp/RaspStore.hpp"
-#include "Weather/Rasp/Providers.hpp"
-#include "ui/control/List.hpp"
-#include "Form/Edit.hpp"
```
Comment (MaxKellermann) [inline src/Dialogs/Weather/RASPDialog.cpp:34]: Why remove this?

File: src/Dialogs/Weather/RASPDialog.cpp [hunk 7]
```diff
@@ -71,7 +62,6 @@ class RASPSettingsPanel final
   void UpdateTimeControl() noexcept;
   void OnTimeModified(const DataFieldEnum &df) noexcept;
   void Download() noexcept;
-
```
Comment (MaxKellermann) [inline src/Dialogs/Weather/RASPDialog.cpp:74]: Whitespace change

File: src/Dialogs/Weather/RASPDialog.cpp [hunk 8]
```diff
@@ -127,7 +117,6 @@ RASPSettingsPanel::UpdateTimeControl() noexcept
         _stprintf(timetext, _T("%02u:%02u"), t.hour, t.minute);
         time_df.addEnumText(timetext, t.GetMinuteOfDay());
       });
-
```
Comment (MaxKellermann) [inline src/Dialogs/Weather/RASPDialog.cpp:130]: Whitespace change

File: src/Dialogs/Weather/RASPDialog.cpp [hunk 9]
```diff
@@ -244,7 +186,6 @@ CreateRaspWidget() noexcept
   auto rasp = DataGlobals::GetRasp();
   return std::make_unique<RASPSettingsPanel>(std::move(rasp));
 }
-
```
Comment (MaxKellermann) [inline src/Dialogs/Weather/RASPDialog.cpp:247]: Whitespace change

**Review by MaxKellermann — CHANGES_REQUESTED** [review]

File: src/Repository/FileType.hpp (hunk 1 above)
Comment (lordfolken) [reply src/Repository/FileType.hpp]: Comma added

**Pull request merged**